include/mysql/psi/mysql_socket.h

```diff
@@ -562,6 +562,12 @@ inline_mysql_socket_socket
       (key, (const my_socket*)&mysql_socket.fd, NULL, 0);
   }
 #endif
+
+  /* SOCK_CLOEXEC isn't always a number - can't preprocessor compare */
+#if defined(HAVE_FCNTL) && defined(FD_CLOEXEC) && !defined(HAVE_SOCK_CLOEXEC)
+  (void) fcntl(mysql_socket.fd, F_SETFD, FD_CLOEXEC);
+#endif
+
```

Comment:
Why not something like
```
    unix_sock= mysql_socket_socket(key_socket_unix, AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0);
```

Reply:
osx and solaris looking at the man pages - https://www.unix.com/man-page/osx/2/socket/
https://docs.oracle.com/cd/E36784_01/html/E36875/socket-3socket.html#REFMAN3Bsocket-3socket

Reply:
Thanks! Makes sense.